plugins/catalog-backend/src/service/DefaultRefreshService.ts

```diff
@@ -45,4 +49,9 @@ export class DefaultRefreshService implements RefreshService {
       });
     });
   }
+  async refreshByRefreshKeys(options: RefreshByRefreshKeysOptions) {
+    await this.database.transaction(async tx => {
+      await this.database.refreshByRefreshKeys(tx, options);
+    });
+  }
```

Comment:
I feel we need to do something different here tbh, just passing through without authorization is pretty unexpected. What I'm thinking is that we'd keep refreshing by keys internal, so it's not a thing you can request via the API. The only way to refresh via keys would be some form of direct runtime access for example via the entity providers API.

Have you considered this step yet? didn't see it in the RFC. How do you end up wanting to pipe refresh data into the catalog? Maybe we could even skip this part in the PR and leave it to a followup

Reply:
Yea, I agree my initial idea was to just expose it on the refresh service and have its own endpoint. I can accept that we do not want to do this.

The other thing that seems kinda easy is to add the refreshing to the EntityProvider, they have access to the DefaultProcessingDatabase so should be easy to expose a new method that calls the appropriate refresh function. However, I don't think EntityProviders should handle refreshes, for me they should care about creating/deleting the entities. So I'd probably not put this behaviour on to the EntityProviders. 

I think refreshing belongs to the catalog. So I am thinking about something like that the CatalogBuilder could return a refresh function and then everyone could decide how they want to use that to refresh their stuff. 

stg like:
```ts
export default async function createPlugin(
  env: PluginEnvironment,
): Promise<Router> {
  const builder = await CatalogBuilder.create(env);
  builder.addProcessor(new ScaffolderEntitiesProcessor());
  const { processingEngine, router, refresh } = await builder.build();
  
  await processingEngine.start();
  
  refresh(['refreshKey-1', 'refreshKey-2'])

  return router;
}

```

Reply:
I actually removed the possibility to trigger refreshes via the RefreshService to be able to tackle it in a separate PR if we decide it that way.